agent/config/deprecated.go

```diff
@@ -34,11 +34,21 @@ func applyDeprecatedConfig(d *decodeTarget) (Config, []string) {
 	dep := d.DeprecatedConfig
 	var warns []string
 
+	// TODO(boxofrad): The DeprecatedConfig struct only holds fields that were once
+	// on the top-level Config struct (not nested fields e.g. ACL.Tokens) maybe we
+	// should rethink this a bit?
```

Comment:
What do you have in mind?

I think this pattern of normalizing in `applyDeprecatedConfig` is still good, because it means our merge config logic can ignore the deprecated fields. Even if the fields are still accessible to that logic.  

I left a suggestion above on how we might be able to make nested deprecated fields more obviously deprecated.

Reply:
Totally agree with keeping `applyDeprecatedConfig` 💯 

My main concern is that we've lost the clean separation of all the deprecated stuff living on `DeprecatedConfig` (which is embedded in `decodeTarget` rather than `Config` itself) because of the fields on nested structs.

It'd be cool if we could keep `ACL.Tokens`'s deprecated fields on `DeprecatedConfig` somehow. Maybe we could decode `Config` and `DeprecatedConfig` separately and merge them together — but I think it would complicate things like handling unrecognised field names. 🤷🏻